### PR TITLE
FIX: sk3 wizard regressions

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: app/assets/javascripts/select-kit/addon/components/select-kit.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default Component.extend(
       this.set(
         "selectKit",
         EmberObject.create({
-          uniqueID: this.attrs?.id || guidFor(this),
+          uniqueID: this.attrs?.id?.value || this.attrs?.id || guidFor(this),
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit.js:80]: prevents the case where we get a mutable cell

File: app/assets/javascripts/select-kit/addon/components/select-kit.js [hunk 2]
```diff
@@ -836,7 +836,7 @@ export default Component.extend(
       this.clearErrors();
 
       const inModal = this.element.closest("#discourse-modal");
-      if (inModal && this.site.mobileView) {
+      if (inModal && this?.site?.mobileView) {
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit.js:839]: site is not defined in wizard

File: app/assets/javascripts/select-kit/addon/components/select-kit.js [hunk 3]
```diff
@@ -868,7 +868,7 @@ export default Component.extend(
           `#${this.selectKit.uniqueID}-body`
         );
 
-        const placementStrategy = this.site.mobileView ? "absolute" : "fixed";
+        const placementStrategy = this?.site?.mobileView ? "absolute" : "fixed";
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit.js:871]: site is not defined in wizard

File: app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js [hunk 4]
```diff
@@ -36,7 +35,7 @@ export default Component.extend(UtilsMixin, {
   didInsertElement() {
     this._super(...arguments);
 
-    if (!this.site.mobileView) {
+    if (!this?.site?.mobileView) {
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js:38]: site is not defined in wizard

File: app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js [hunk 5]
```diff
@@ -45,7 +44,7 @@ export default Component.extend(UtilsMixin, {
 
   willDestroyElement() {
     this._super(...arguments);
-    if (!this.site.mobileView && this.element) {
+    if (!this?.site?.mobileView && this.element) {
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js:47]: site is not defined in wizard

File: app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js [hunk 6]
```diff
@@ -115,9 +114,13 @@ export default Component.extend(UtilsMixin, {
     return this.getValue(this.selectKit.highlighted);
   }),
 
-  isHighlighted: propertyEqual("rowValue", "highlightedValue"),
+  isHighlighted: computed("rowValue", "highlightedValue", function () {
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js:117]: import { propertyEqual } from "discourse/lib/computed"; is not defined in wizard

File: app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js [hunk 7]
```diff
@@ -115,9 +114,13 @@ export default Component.extend(UtilsMixin, {
...
+    return this.rowValue === this.highlightedValue;
+  }),
 
-  isSelected: propertyEqual("rowValue", "value"),
+  isSelected: computed("rowValue", "value", function () {
```
Comment (jjaffeux) [inline app/assets/javascripts/select-kit/addon/components/select-kit/select-kit-row.js:121]: import { propertyEqual } from "discourse/lib/computed"; is not defined in wizard

File: app/assets/stylesheets/common/select-kit/select-kit.scss [hunk 8]
```diff
@@ -262,6 +263,8 @@
       border-radius: 0;
       box-shadow: none;
       width: 100%;
+      box-sizing: border-box;
```
Comment (jjaffeux) [inline app/assets/stylesheets/common/select-kit/select-kit.scss:266]: ensures we have a min height as wizard css doesn't have input styles

**Review by pmusaraj — APPROVED** [review]

**Pull request merged**